torch/csrc/utils/tensor_new.cpp

```diff
@@ -225,6 +225,12 @@ void recursive_store(char* data, IntArrayRef sizes, IntArrayRef strides, int64_t
 
   PyObject** items = PySequence_Fast_ITEMS(seq.get());
   for (int64_t i = 0; i < n; i++) {
+    if (PyArray_Check(items[i])) {
```

Comment:
I'm not entirely sure if this check will _only_ return true if the item is a `numpy.ndarray`. Perhaps it will be true for other types as well?

Reply:
Answer: `PyArray_Check` is a NumPy function, so it should only be true if the item is a `numpy.ndarray`, as far as I understand: https://numpy.org/doc/1.17/reference/c-api.array.html#c.PyArray_Check